### ISSUE TITLE
Clicking on banners

### ISSUE THREAD

**Issue by nrshapiro** [Psychoanalytic-Electronic-Publishing/PEP-Web-User-Interface#475]
Clicked on the New Library of Psychoanalysis banner on a book, it took me here:

https://stage.pep-web.rocks/browse/NLP/volumes?openNotificationModal=false

No banner, no title, just a list of volumes.

![image](https://user-images.githubusercontent.com/4837554/116592588-29afc500-a8ee-11eb-8a80-6e79a35a342c.png)

Inspect shows image name is null.

**Comment by bakerac4** [comment]
@nrshapiro a few questions on this one.

1. I cant find NLP in the journal sidebar list on the browse page (the only way I can find it is to go there directly?)
2. The reason there is no image is because the backend is returning no data from `https://stage-api.pep-web.rocks/v2/Metadata/Journals/?sourcecode=NLP`. Should it be?

**Comment by nrshapiro** [comment]
@bakerac4 

NLP is not a journal, it's a book series, e.g., https://pep-web.rocks/browse/document/NLP.001.0001A

As you can see, there's a banner there.   Click on it.

This URL is a vestige of your earlier builds, but since it's there, and useful, it should also have the banner for NLP, like the NLP books do.

https://api.pep-web.rocks/v2/Documents/Image/bannerNLPLogo.gif/

The same will be true of the other book series, IPL.

go here:  https://stage.pep-web.rocks/browse/document/IPL.115.0001A?page=P0001

Click on the banner, takes you here:

https://stage.pep-web.rocks/browse/IPL/volumes?openNotificationModal=false

Also, no logo there.  

Classic books (ZBK) does the same thing actually.

Go here:  https://stage.pep-web.rocks/browse/document/ZBK.054.0000A

Click on the banner, takes you here: 

https://stage.pep-web.rocks/browse/ZBK/volumes?openNotificationModal=false

No banner.

There is an issue though stopping you from getting the banner name from the API even in the api/metadata/books call.  Despite the interactive doc saying:

```
The data is pulled from the database ISSN table. Subvolumes of a book series (e.g., GW, SE) are not returned, nor is any volume marked with multivolumesubbok in the src_type_qualifier column. This is exactly what's currently in PEP-Web's presentation today.

To get metadata for a specific book of a series or subvolume of a multivolume book like GW, SE, NLP, IPL, or ZBK, use the first two parts of the source document ID (series and series volume) without the period separator used in the document ID. For example, to request metadata for SE volume 6, use SE006 in the sourcecode API parameter. Similarly, use NLP014 to get metadata on the book published as volume 14 of the NLP series. While ZBK is a general series ID for classic books, the same is true...use ZBK047 to return metadata on that specific book.
```

Its inaccurate.  There's no no data similar to what I put in for SE and GW series for the NLP, ZBK, and IPL basecodes.  However, I just added bookserieshead (dummy) records for NLP, IPL, and ZBK to stage, as I did for SE/GW, so now you'd get a banner name if you search for that source code, like:

https://stage-api.pep-web.rocks/v2/Metadata/Books/?sourcecode=ZBK
https://stage-api.pep-web.rocks/v2/Metadata/Books/?sourcecode=NLP
and
https://stage-api.pep-web.rocks/v2/Metadata/Books/?sourcecode=ZBK

We'll have to test and make sure that doesn't mess up the client in other ways, because there's no real books at the IPL.000.0000A, etc and of course, some of the book information is not applicable to a book series head (and I'll still have to figure out others, like year range.)

Once that's proven, and finished, I'll also move the metadata to the production database.

**Comment by bakerac4** [comment]
![Screen Shot 2021-05-04 at 10 32 35 AM](https://user-images.githubusercontent.com/9952936/117020042-0c9b3d80-acc4-11eb-962f-34d41b79edf7.png)

@nrshapiro when Im building link out, how do I know if it goes to a journal or a book? I need to differentiate at that point if I should be going to /browse/journal/IJP/volumes or /browse/book/NLP/volumes

**Comment by nrshapiro** [comment]
@bakerac4 

You know because the link the client is generating for the banner its placing, for example in the picture above, is a book, right?

![image](https://user-images.githubusercontent.com/4837554/117030593-ca76f980-accd-11eb-9d22-1e5a774fea79.png)

and in the XML itself:

![image](https://user-images.githubusercontent.com/4837554/117030905-145fdf80-acce-11eb-9578-2f40600434ef.png)

**Comment by bakerac4** [comment]
@nrshapiro Im now seeing the following in the book list.

![Screen Shot 2021-05-04 at 2 58 05 PM](https://user-images.githubusercontent.com/9952936/117055428-23a05680-ace9-11eb-83b1-67c904b5329f.png)

**Comment by bakerac4** [comment]
Also are there any embargoes for books?

**Comment by nrshapiro** [comment]
@bakerac4 

>>Also are there any embargoes for books?
No. 

Regarding the prior comment: 

I've been booked solid all day between meetings and juggling incoming requests, so I didn't have a chance to look at consequences of adding that data, but I'll take a look by tonight.   

I'm fairly confident I can make sure those are filtered out in the book listing endpoint, since they are clearly indicated in the database as "top level" multivolume book info.  In any case, the original banner link issue isn't serious enough to derail the Beta, and if I don't push that data to production, the extra listings won't be there anyway.  (We are still planning to put the beta out tomorrow.)

**Comment by bakerac4** [comment]
@nrshapiro the other thing we can do (because this is sort of a large change on my side) is just turn off the linking of the banner when its a book so you can't get to that page.

**Comment by nrshapiro** [comment]
@bakerac4 

This is fixed on stage via a workaround in the server.  The workaround was basically just an "extreme" version of removing data from the database fields for those unused PEP source codes.  I'm not sure why it behaves so differently than just adjusting the database itself to return info on those items, but it returns minimal information for those, and appears to solve the problem.  It likely has to do with what fields you are using to decide what to put in the book list, or the change in source type I used (the database currently doesn't support anything but the standard names).

e.g., for NLP

{
  "sourceInfo": {
    "responseInfo": {    ...    },
    "responseSet": [
      {
        "sourceType": "book series",
        "PEPCode": "NLP",
        "bannerURL": "https://stage-api.pep-web.rocks/v2/Documents/Image/bannerNLPLogo.gif"
      }
    ]
  }
}

HOWEVER: I do think it's a good idea to turn these off  When I was testing last night, **I noticed that this applies to EVERY video,** and I didn't add the workaround for those.    E.g., the link on [this video](https://stage.pep-web.rocks/browse/document/IPSAVS.001A.0016A) takes you here:

https://stage.pep-web.rocks/browse/IPSAVS/volumes?openNotificationModal=false

Clicking on the video header to get an index of the one volume you were looking at doesn't make any sense anyway, so these should at a minimum be removed.  

But the books links, even though they works now, don't make sense either, so YES, probably should just remove them all.
